PureBasicIDE/ColorSchemes.pb

```diff
@@ -0,0 +1,324 @@
+          
+          If (*ColorScheme) ; struct already specified - part of InitColorSchemes() list
+            ; Intentionally overwrite schemes of existing names - allows you to tweak the default themes, if desired
+            RemoveColorSchemeIfExists(Name)
+          Else ; NULL --> dynamically allocate a struct now - NOT part of InitColorSchemes() list!
+            *ColorScheme = AllocateStructure(ColorSchemeStruct)
+          EndIf
+          
+          If (*ColorScheme)
+            With *ColorScheme
+              \Name = Name
+              \File = File
+              
+              ; Load all defined colors into map...
+              For i = 0 To #COLOR_Last_IncludingToolsPanel
+                \ColorValue[i] = #ColorSchemeValue_Undefined
+                ColorValueString.s = ReadPreferenceString(ColorName(i), "")
+                If (ColorValueString <> "")
+                  If (ReadPreferenceLong(ColorName(i) + "_Used", 1) = 1)
+                    If (FindString(ColorValueString, "RGB", 1, #PB_String_NoCase))
+                      \ColorValue[i] = ColorFromRGBString(ColorValueString)
+                    Else
+                      \ColorValue[i] = Val(ColorValueString) & $00FFFFFF
+                    EndIf
+                  EndIf
+                EndIf
+              Next i
+              Result = *ColorScheme
+              
+            EndWith
+          EndIf
+          
+        EndIf
+      EndIf
+      ClosePreferences()
+    EndIf
+  EndIf
+  
+  ProcedureReturn (Result)
+EndProcedure
+
+; Initialize color names, built-in color schemes, and external found color schemes
+Procedure InitColorSchemes()
+  
+  ; Only need to initialize color schemes once
+  If (NbSchemes > 0)
+    ProcedureReturn
+  EndIf
+  
+  ; Read color key names into indexable array
+  Restore ColorKeys
+  For i = 0 To #COLOR_Last
+    Read.s ColorName(i)
+  Next i
+  ColorName(#COLOR_ToolsPanelFrontColor) = "ToolsPanel_FrontColor"
+  ColorName(#COLOR_ToolsPanelBackColor)  = "ToolsPanel_BackColor"
+  
+  
+  ; First, load embedded DataSection default color schemes
+  ClearList(ColorScheme())
+  Restore DefaultColorSchemes
+  Read.s Name.s
+  While (Name <> "")
+    AddElement(ColorScheme())
+    ColorScheme()\Name = Name
+    ReadColorSchemeFromDataSection(@ColorScheme())
+    If (ListIndex(ColorScheme()) = 0)
+      ColorScheme()\IsIDEDefault = #True
+    EndIf
+    Read.s Name
+  Wend
+  NbSchemes = ListSize(ColorScheme())
+  
+  ; Then, scan 'themes' subfolder!
+  If (PureBasicPath$)
+    Dir = ExamineDirectory(#PB_Any, PureBasicPath$ + #DEFAULT_ThemePath, "*")
+    If (Dir)
+      While (NextDirectoryEntry(Dir))
+        If (DirectoryEntryType(Dir) = #PB_DirectoryEntry_File)
+          File.s = PureBasicPath$ + #DEFAULT_ThemePath + #PS$ + DirectoryEntryName(Dir)
+          Select (LCase(GetExtensionPart(File)))
+            Case "prefs", "scheme", "theme", "ini" ; only attempt to load certain file extensions
```

Comment:
Why allow so many different file extensions? We define the format for this, so we can also define what extension is to be used. In my opinion, `prefs` alone is fine.

Reply:
Again just trying to follow the "strict in what output you generate, but lenient in what input you accept" style. Yes it could only accept `prefs`. I thought at least one other extension (`scheme` ?) would be nice so users could differentiate standalone color scheme files from true "IDE preferences" files.